semgrep-core/bin/Main.ml

```diff
@@ -843,8 +870,25 @@ let main () =
     spf "Usage: %s [options] <pattern> <files_or_dirs> \nOptions:"
       (Filename.basename Sys.argv.(0))
   in
+
+  let argv =
+   (Array.to_list Sys.argv) @
+   (if Sys.getenv_opt "SEMGREP_CORE_DEBUG" <> None then ["-debug"] else[])@
+   (if Sys.getenv_opt "SEMGREP_CORE_PROFILE" <> None then ["-profile"] else[])
+  in
+
   (* does side effect on many global flags *)
-  let args = Common.parse_options (options()) usage_msg Sys.argv in
+  let args = Common.parse_options (options()) usage_msg (Array.of_list argv) in
+
+  if !debug then begin
+    pr2 "Debug mode On";
+    pr2 (spf "Executed as: %s" (Sys.argv|>Array.to_list|> String.concat " "));
+  end;
+  if !profile then begin
```

Comment:
@rcoh maybe this was the issue. Maybe you were running the ocaml programs with profiling information but
because of -j the job was actually done in another process ...

Reply:
Yeah that occurred to me after I read that multi threading in OCaml is actually multiprocessing

Reply:
Well, OCaml has concurrent threads (Xavier Leroy the author of OCaml actually added the first POSIX C thead library for Linux a long time ago, and he did it because he wanted threads in OCaml), but it does not have yet multi-core threads. There is work ongoing to suppor that.
Note that neither Python/PHP/Ruby/... have multi-core threads either.